src/Search/SearchHandler.php

```diff
@@ -81,14 +81,21 @@ public function search(AdminInterface $admin, $term, $page = 0, $offset = 20)
 
         $datagrid = $admin->getDatagrid();
 
+        $datagridValues = $datagrid->getValues();
+
         $found = false;
         foreach ($datagrid->getFilters() as $filter) {
-            /** @var $filter FilterInterface */
+            /** @var FilterInterface $filter */
+            $formName = $filter->getFormName();
+
             if ($filter->getOption('global_search', false)) {
                 $filter->setOption('case_sensitive', $this->caseSensitive);
                 $filter->setCondition(FilterInterface::CONDITION_OR);
-                $datagrid->setValue($filter->getFormName(), null, $term);
+                $datagrid->setValue($formName, null, $term);
                 $found = true;
+            } elseif (isset($datagridValues[$formName])) {
```

Comment:
I don't have in mind if it is possible, but what if the filter has `null` as default value?

Reply:
Well, if a value is null, it means it isn't set. That's why I set it to null.

Reply:
Well, I'm referring to cases where a default filter could be configured like `$filterValues['expiryDate']['value'] = null;`, but if this case has no conflicts with your changes, it's fine for me.